modules/internal/ChapelArray.chpl

```diff
@@ -2744,6 +2744,9 @@ module ChapelArray {
 
       reallocateArray(newRange, debugMsg="push_back reallocate");
 
+      // This could "move" from val to the array element, but
```

Comment:
I don't understand this comment. I think that `this[this.domain.high]` should effectively be uninitialized memory.  It was allocated by `reallocateArray()` above (likely as part of a previous allocation for several not-yet-used elements) and is not used until this assignment.

Reply:
Thanks for explaining.  I'm OK with just doing 1. for now.